tests/sycl/group_functions.cpp

```diff
@@ -0,0 +1,1800 @@
+
+            BOOST_TEST(detail::TCompare(expected, computed), detail::T2str(computed) << " at position " << i << " instead of " << detail::T2str(expected) << " for case: id");
+
+            if (!detail::TCompare(expected, computed))
+              break;
+          }
+        };
+
+        test_nd_group_function_1d<__LINE__, T>(local_size, global_size,
+                                               offset_margin, offset_divisor,
+                                               buffer_size, dataGenerator,
+                                               functionUnderTest, testFunction);
+    }
+}
+#endif
+*/
+BOOST_AUTO_TEST_CASE(group_x_of_local) {
+    using T = char;
+
+    const size_t local_size       = 256;
+    const size_t global_size      = 1024;
+    const size_t local_size_x     = 16;
+    const size_t local_size_y     = 16;
+    const size_t global_size_x    = 32;
+    const size_t global_size_y    = 32;
+    const size_t offset_margin    = global_size;
+    const size_t offset_divisor   = 1;
+    const size_t buffer_size      = global_size;
+    const auto dataGenerator      = [](std::vector<T> &v){
+        detail::create_bool_test_data(v, local_size, global_size);
+    };
+
+    {
+        const auto functionUnderTest  = [](auto acc, size_t global_linear_id, sycl::sub_group sg, auto g, T local_value){
+          acc[global_linear_id] = sycl::group_any_of(g, static_cast<bool>(local_value));
+        };
+        const auto testFunction       = [](const std::vector<T> &vIn, const std::vector<T> &vOrig){
+            detail::check_binary_reduce<T, __LINE__>(vIn, local_size, global_size, std::vector<bool>{true, false, true, true}, "any_of");
+        };
+
+        test_nd_group_function_1d<__LINE__, T>(local_size, global_size,
+                                               offset_margin, offset_divisor,
+                                               buffer_size, dataGenerator,
+                                               functionUnderTest, testFunction);
+
+        test_nd_group_function_2d<__LINE__, T>(local_size_x, local_size_y,
+                                               global_size_x, global_size_y,
+                                               offset_margin, offset_divisor,
+                                               buffer_size, dataGenerator,
+                                               functionUnderTest, testFunction);
+    }
+
+    {
+        const auto functionUnderTest  = [](auto acc, size_t global_linear_id, sycl::sub_group sg, auto g, T local_value){
+          acc[global_linear_id] = sycl::group_all_of(g, static_cast<bool>(local_value));
+        };
+        const auto testFunction       = [](const std::vector<T> &vIn, const std::vector<T> &vOrig){
+            detail::check_binary_reduce<T, __LINE__>(vIn, local_size, global_size, std::vector<bool>{false, false, false, true}, "all_of");
+        };
+
+        test_nd_group_function_1d<__LINE__, T>(local_size, global_size,
```

Comment:
It's an interesting idea to use `__LINE__` as kernel name. I'm torn between thinking that it's genius or evil C-macro hacking ;)

Reply:
Since I call the kernel in the function I need some way to get unique names, the type alone is not enough. If you have another idea I'll glady use it. Once we use c++20 this can be replaced with [std::source_location::current()](https://en.cppreference.com/w/cpp/utility/source_location/current).

Reply:
The traditional way is to just name it: `test_nd_group_function_1d<class my_test_case_name_or_a_description_of_it, T>(...)`. But I see that this can be cumbersome when there are many test cases.

Reply:
I Use line to make the tests as copy&past ready as possible. As an added bonus I get the line number if something goes wrong and I have to read mangled names :>